From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Overwrite987 <artem.korolev_2003@mail.ru>
Date: Mon, 28 Jul 2025 03:35:42 +0300
Subject: [PATCH] Do not create fire if explosion was cancelled

Related to PaperMC/Paper#12441 and PaperMC/Paper#11840.
Adapted from Leaf.


diff --git a/net/minecraft/world/level/ServerExplosion.java b/net/minecraft/world/level/ServerExplosion.java
--- a/net/minecraft/world/level/ServerExplosion.java
+++ b/net/minecraft/world/level/ServerExplosion.java
@@ -663,6 +663,7 @@ public class ServerExplosion implements Explosion {
     }
 
     private void createFire(final List<BlockPos> targetBlocks) {
+        if (this.wasCanceled) return; // Folia performance pack - Leaf
         for (BlockPos pos : targetBlocks) {
             if (this.level.random.nextInt(3) == 0 && this.level.getBlockState(pos).isAir() && this.level.getBlockState(pos.below()).isSolidRender()) {
                 // CraftBukkit start - Ignition by explosion
